### PR TITLE
Setup external DNS servers after daemon restart with live-restore

### PR DESCRIPTION
**Pull request by sanimej**
Signed-off-by: Santhosh Manohar <santhosh@docker.com>

### CODE REVIEW

File: sandbox_store.go [hunk 1]
```diff
@@ -240,6 +246,10 @@ func (c *controller) sandboxCleanup(activeSandboxes map[string]interface{}) {
 			heap.Push(&sb.endpoints, ep)
 		}
 
+		for _, dns := range sbs.ExtDNS {
```
Comment (aboch) [inline sandbox_store.go]: Why not initializing the extDNS field above in the sb initialization ?

```
...
extDNS: sbs.extDNS,
...
```

File: sandbox_dns_unix.go [hunk 2]
```diff
@@ -31,12 +31,19 @@ func (sb *sandbox) startResolver() {
...
+		// external DNS servers from the restored sandbox for embedded
+		// server to use.
+		if !restore {
+			err = sb.rebuildDNS()
+			if err != nil {
+				log.Errorf("Updating resolv.conf failed for container %s, %q", sb.ContainerID(), err)
+				return
+			}
+		} else {
+			sb.resolver.SetExtServers(sb.extDNS)
```
Comment (aboch) [inline sandbox_dns_unix.go]: Is it ok to skip it now in the regular path, no restore ?
Before it was being executed.

File: sandbox_store.go (hunk 1 above)
Comment (sanimej) [inline sandbox_store.go]: Isn't it better to recreate the slice in the new sandbox ? holding on to the reference of sbs.extDNS would prevent  sbState from being GC'ed.

File: sandbox_dns_unix.go (hunk 2 above)
Comment (sanimej) [inline sandbox_dns_unix.go]: It shouldn't be skipped in the regular path. Thanks.

**Comment by aboch** [comment]
LGTM

**Pull request merged**